src/python/pants/engine/goal.py

```diff
@@ -44,69 +30,63 @@ def deprecated_cache_setup_removal_version(cls):
     """
     return None
 
+  @classproperty
+  def options_scope(cls):
+    return cls.name
+
   @classmethod
-  def register_options(cls, register):
-    """Register options for the `Goal.Options` of this `Goal`.
+  def subsystem_dependencies(cls):
+    # NB: `Goal.Options` implements `SubsystemClientMixin` in order to allow v1 `Tasks` to
```

Comment:
`s/Goal.Options/GoalSubsystem/`

Reply:
I think most of these comments are on the first commit, where I had not yet updated documentation. The documentation should be good to go when looking at the whole diff. Sorry for the confusion!